### ISSUE TITLE
Consider ProfileConfigFileAWSCredentialsProvider profiles compatible with boto3

### ISSUE THREAD

**Issue by TTimo** [aws/aws-sdk-cpp#845]
### What platform/OS are you using?
Windows
### What compiler are you using? what version?
Visual Studio 15 2017, SDK 1.4.33

It seems the boto3 syntax for profiles in config files works like this:

```
[profile "my profile"]
aws_access_key_id=...
```

And for a config profile to work with `ProfileConfigFileAWSCredentialsProvider`, it needs to be written like this:

```
[my profile]
aws_access_key_id=...
```

Currently my config files have two entries with the same keys, just so each tool will find the profile. It would be nice if that could be unified.

**Comment by wps132230** [comment]
I think both `[profile name]` and `[name]` work for providing credentials (and I tested it).

Can you provide the details in your `~/.aws/credentials` and `~/.aws/config`? (sanitize any sensitive information)

**Comment by TTimo** [comment]
Only `[my profile]` works for me. I just ran into this again on a new system.

I only use a `%USERPROFILE%/.aws/config`, no `%USERPROFILE%/.aws/credentials` present.

```
[profile "my profile"]
aws_access_key_id=mykeyid
aws_secret_access_key=mysecret
region=us-east-2

[my profile]
aws_access_key_id=mykeyid
aws_secret_access_key=mysecret
region=us-east-2
```

I have to have the `[profile "my profile"]` for boto3 scripts, and `[my profile]` for anything using the C++ sdk. If I comment out the `[my profile]` section the credentials code fails.

`Aws::Auth::ProfileConfigFileAWSCredentialsProvider CredentialsProvider("my profile");`

Now that I think of it, I haven't tried `Aws::Auth::ProfileConfigFileAWSCredentialsProvider CredentialsProvider("profile \"my profile\"");` .. that's goofy but it might work.

**Comment by KaibaLopez** [comment]
Hey @TTimo, trying to do a little backlog clearing and wanted to ask, did you end up trying your change? was your problem solved or are there any other doubts you've got regarding this?

**Comment by TTimo** [comment]
I'll give it another spin today.

**Comment by TTimo** [comment]
Weeelll I might still be around to test it but that'll take longer than anticipated.

**Comment by TTimo** [comment]
The 'workaround' doesn't work. It wouldn't be adviseable either anyway.

tl;dr - boto3 supports putting credentials in `config` but requires a different syntax (`[profile "my profile"]`). This is not compatible with aws-sdk-cpp.

It's best to just put credentials in the `credentials` file where the syntax `[my profile]` will work for both boto3 and aws-sdk-cpp

**Comment by KaibaLopez** [comment]
Thanks @TTimo !
